src/page-modules/assistant/client/journey-planner/index.ts

```diff
@@ -1,16 +1,59 @@
-import { TripData } from '../../server/journey-planner/validators';
-import { TripQuery } from '../../types';
+import useSWR from 'swr';
+import { NonTransitTripData, TripQuery, TripData } from '../../types';
+import { swrFetcher } from '@atb/modules/api-browser';
+import useSWRInfinite from 'swr/infinite';
 
 export type TripApiReturnType = TripData;
+export type NonTransitTripApiReturnType = NonTransitTripData;
 
-export async function nextTripPatterns(
-  query: TripQuery,
-): Promise<TripApiReturnType> {
-  const queryString = tripQueryToQueryString(query);
+function createKeyGetterOfQuery(query: TripQuery) {
+  return (pageIndex: number, previousPageData: TripApiReturnType) => {
+    const cursorKey =
+      query.searchMode === 'arriveBy' ? 'previousPageCursor' : 'nextPageCursor';
+
+    if (previousPageData && !previousPageData[cursorKey]) return null;
+
+    const queryString = tripQueryToQueryString(query);
+
+    // first page, we don't have `previousPageData`
+    if (pageIndex === 0) return `/api/assistant/trip?${queryString}`;
+
+    return `/api/assistant/trip?cursor=${previousPageData[cursorKey]}&${queryString}`;
+  };
+}
+
+export function useTripPatterns(query: TripQuery) {
+  const { data, error, isLoading, isValidating, size, setSize } =
+    useSWRInfinite<TripApiReturnType>(
+      createKeyGetterOfQuery(query),
+      swrFetcher,
+      {},
+    );
 
-  const result = await fetch(`/api/assistant/trip?${queryString}`);
+  const isLoadingInitialData = !data && isLoading;
+  const isLoadingMore = isValidating && size > 1;
+  return {
+    trips: data,
+    isLoading: isLoadingInitialData,
+    isError: error,
+    loadMore: () => setSize(size + 1),
+    isLoadingMore,
+  };
+}
+
+export function useNonTransitTrip(query: TripQuery) {
+  const queryString = tripQueryToQueryString(query);
+  const { data, error, isLoading } = useSWR<NonTransitTripApiReturnType>(
```

Comment:
Not using `useSWRInfinite` here?

Reply:
No, that is unnecessary since we are only after one example of the non-transit trip types (no pagination or infinite scroll here)

Reply:
Won't revalidation on focus/mount etc come into play?

Reply:
Not sure what you are thinking of 🤔 `useSWRInfinite` vs. `useSWR` should not behave differently regarding that